apps/smtp/src/modules/smtp/configuration/smtp-metadata-manager.ts

```diff
@@ -1,8 +1,12 @@
 import { SettingsManager } from "@saleor/app-sdk/settings-manager";
-import { SmtpConfig } from "./smtp-config-schema";
-import { fromAsyncThrowable, fromPromise, fromThrowable, ok, ResultAsync } from "neverthrow";
+import { fromPromise, fromThrowable, ok, ResultAsync } from "neverthrow";
+
 import { BaseError } from "../../../errors";
+import { racePromise } from "../../../lib/race-promise";
 import { createLogger } from "../../../logger";
+import { SmtpConfig } from "./smtp-config-schema";
+
+const PULL_CONFG_TIMEOUT = 3000;
```

Comment:
would be good if you add some comment, that this is an escape hatch in case of metadata is too slow, etc, so we know:
1. why
2. why 3000